packages/plugin/src/Integrations/PaymentGateways/PayPal/Services/PayPalPriceService.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Solspace\Freeform\Integrations\PaymentGateways\PayPal\Services;
+
+use Solspace\Freeform\Form\Form;
+use Solspace\Freeform\Integrations\PaymentGateways\PayPal\Fields\PayPalField;
+
+class PayPalPriceService
+{
+    public function getAmount(Form $form, PayPalField $field): int
+    {
+        $amount = $field->getAmount();
+
+        switch ($field->getAmountType()) {
+            case PayPalField::AMOUNT_TYPE_DYNAMIC:
+                $amount = $this->getDynamicAmount($form, $field);
+
+                break;
+
+            case PayPalField::AMOUNT_TYPE_FIXED:
+            default:
+                $amount = $field->getAmount();
+
+                break;
+        }
+
+        $finalAmount = (int) round($amount * 100);
+
+        if ($finalAmount <= 0) {
+            throw new \Exception("PayPal amount must be greater than 0. Current amount: {$amount} (final: {$finalAmount})");
+        }
+
+        return $finalAmount;
+    }
+
+    private function getDynamicAmount(Form $form, PayPalField $field): float
+    {
+        $amount = 0;
+
+        $amountField = $field->getAmountField();
+        if ($amountField) {
+            $value = $amountField->getValue();
+            if (is_numeric($value)) {
+                $amount = (float) $value;
+            } elseif (\is_string($value)) {
+                $amount = (float) preg_replace('/[^0-9.]/', '', $value);
+            } elseif (\is_array($value)) {
+                $flat = array_filter(array_map(static fn ($v) => is_numeric($v) ? (float) $v : 0, $value));
+                $amount = (float) array_sum($flat);
+            }
+        }
+
+        return (float) $amount;
```

Comment:
- `$form` is never used.
- `$field` cannot be null, so no need to check if it's present and return 0 just yet.
- the less nested IFs are used, the easier the code is to read and maintain down the road. this can be solved by returning the result in each conditional separately, and avoiding `else` statements. 
- if you're doing nested function calls, split them in logical newline blocks, otherwise it's near impossible to figure out what is happening when reading it.

```suggestion
    private function getDynamicAmount(PayPalField $field): float
    {
        $value = $field->getAmountField()->getValue();
        if (is_numeric($value)) {
            return (float) $value;
        }

        if (\is_string($value)) {
            return (float) preg_replace('/[^0-9.]/', '', $value);
        }

        if (\is_array($value)) {
            return (float) array_sum(
                array_filter(
                    array_map(
                        static fn ($val) => is_numeric($val) ? (float) $val : 0,
                        $value
                    )
                )
            );
        }

        return 0;
    }
```